src/app/settings/admin/presets/component.ts

```diff
@@ -102,11 +106,33 @@ export class PresetListComponent implements OnInit, OnDestroy, OnChanges {
     };
 
     merge(of(true), this._presetsChanged)
-      .pipe(switchMap(_ => this._presets$))
-      .pipe(takeUntil(this._unsubscribe))
-      .subscribe(presetList => {
-        this.presets = presetList.items;
+      .pipe(
+        switchMap(_ => this._presets$),
+        tap((presetList: PresetList) => {
+          this.presets = presetList.items;
+        }),
+        switchMap(_ => {
+          const presetStats$ = [];
+          this.presets.forEach((preset: Preset) => {
+            presetStats$.push(this._presetService.getPresetStatsBy(preset.name));
+          });
+          this.isBusyCounter++;
+          return forkJoin(presetStats$);
+        }),
+        map((stats: PresetStat[]) => {
+          const updatePresets = _.cloneDeep(this.presets);
+          updatePresets.forEach((preset: Preset, index: number) => {
+            preset.associatedClusters = stats[index].associatedClusters;
+            preset.associatedClusterTemplates = stats[index].associatedClusterTemplates;
+          });
+          return updatePresets;
```

Comment:
You can skip this map and move it to the subscribe, so cloneDeep() will not be required.

Reply:
@maciaszczykm you are correct it can be skipped and logic can be removed to within in subscribe.

I intentionally added it due following reasons:

- Keep it clear in terms of code readability that we are manipulating existing preset e.g in this case adding filling up values to optional properties

- All manipulations should be done on previous pipe operators and final **subscribe** should only get **latest** data that needs to be bind.


**Note:** let me know your final thoughts i can update it.

thanks

Reply:
You can extract the part from the forEach to the separate method then. Using cloneDeep adds additional complexity that is not required and will slow it down.